src/RA_Condition.h

```diff
@@ -60,46 +74,75 @@ class CompVariable
     };
 
 public:
-    void Set(ComparisonVariableSize nSize, Type nType, unsigned int nInitialValue)
+    inline constexpr CompVariable() noexcept = default;
+    explicit inline constexpr CompVariable(_In_ MemSize nSize,
+                          _In_ CompVariable::Type nType,
+                          _In_ unsigned int nInitialValue) noexcept :
+        m_nVarSize{ nSize },
+        m_nVarType{ nType },
+        m_nVal{ nInitialValue }
+    {
+
+    }
+
+    explicit inline constexpr CompVariable(_In_ unsigned int nInitialValue,
+                                           _In_ unsigned int nPrevVal) noexcept :
+        m_nVal{ nInitialValue },
+        m_nPreviousVal{ nPrevVal }
     {
-        m_nVarSize = nSize;
-        m_nVarType = nType;
-        m_nVal = nInitialValue;
     }
 
-    void SetValues(unsigned int nVal, unsigned int nPrevVal)
+    _CONSTANT_FN Set(_In_ MemSize nSize,
+                     _In_ CompVariable::Type nType,
+                     _In_ unsigned int nInitialValue) noexcept
     {
-        m_nVal = nVal;
-        m_nPreviousVal = nPrevVal;
+        this->swap(CompVariable{ nSize, nType, nInitialValue });
     }
 
-    void ResetDelta()
+    _CONSTANT_FN SetValues(_In_ unsigned int nVal, _In_ unsigned int nPrevVal) noexcept
     {
-        m_nPreviousVal = m_nVal;
+        this->swap(CompVariable{ nVal, nPrevVal });
     }
 
-    bool ParseVariable(const char*& sInString);	//	Parse from string
-    void SerializeAppend(std::string& buffer) const;
+    _CONSTANT_FN ResetDelta() noexcept { m_nPreviousVal = m_nVal; }
+
+    bool ParseVariable(_Inout_ const char*& sInString);	//	Parse from string
+    void SerializeAppend(_Out_ std::string& buffer) const;
 
     unsigned int GetValue();				//	Returns the live value
 
-    inline void SetSize(ComparisonVariableSize nSize) { m_nVarSize = nSize; }
-    inline ComparisonVariableSize Size() const { return m_nVarSize; }
+    _CONSTANT_FN SetSize(MemSize nSize) noexcept { m_nVarSize = nSize; }
+    _NODISCARD _CONSTANT_FN Size() const noexcept { return m_nVarSize; }
 
     _CONSTANT_FN SetType(_In_ Type nType) noexcept { m_nVarType = nType; }
     _NODISCARD _CONSTANT_FN GetType() const noexcept { return m_nVarType; }
 
-    inline unsigned int RawValue() const { return m_nVal; }
-    inline unsigned int RawPreviousValue() const { return m_nPreviousVal; }
+    _NODISCARD _CONSTANT_FN RawValue() const noexcept { return m_nVal; }
+    _NODISCARD _CONSTANT_FN RawPreviousValue() const noexcept { return m_nPreviousVal; }
+
+    inline void swap(_Inout_ CompVariable& a) noexcept
```

Comment:
I'm not convinced of the merit of these methods, but as they're only used by local functions, they should be private.